Assets/BossRoom/Scripts/Client/UI/IPConnectionWindow.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections;
+using TMPro;
+using Unity.Netcode;
+using Unity.Netcode.Transports.UNET;
+using Unity.Netcode.Transports.UTP;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Visual
+{
+    public class IPConnectionWindow : MonoBehaviour
+    {
+        [SerializeField]
+        CanvasGroup m_CanvasGroup;
+
+        [SerializeField]
+        IPUIMediator m_IPUIMediator;
+
+        [SerializeField]
+        TextMeshProUGUI m_TitleText;
+
+        void Awake()
+        {
+            Hide();
+        }
+
+        void Show()
+        {
+            m_CanvasGroup.alpha = 1f;
+            m_CanvasGroup.blocksRaycasts = true;
+        }
+
+        void Hide()
+        {
+            m_CanvasGroup.alpha = 0f;
+            m_CanvasGroup.blocksRaycasts = false;
+        }
+
+        public void ShowConnectingWindow()
+        {
+            var chosenTransport = NetworkManager.Singleton.gameObject.GetComponent<TransportPicker>().IpHostTransport;
+            NetworkManager.Singleton.NetworkConfig.NetworkTransport = chosenTransport;
+
+            void OnTimeElapsed()
+            {
+                PopupManager.ShowPopupPanel("Connection Failed",
+                    "Failed to connect to server and/or invalid network endpoint");
+                Hide();
+                m_IPUIMediator.DisableSignInSpinner();
+            }
+
+            switch (chosenTransport)
+            {
+                case UNetTransport unetTransport:
+                    StartCoroutine(WaitUntilUNETDisconnected(OnTimeElapsed));
+                    break;
+                case UnityTransport unityTransport:
+                    var maxConnectAttempts= unityTransport.MaxConnectAttempts;
+                    var connectTimeoutMS= unityTransport.ConnectTimeoutMS;
+                    StartCoroutine(DisplayUTPReconnectAttempts(maxConnectAttempts, connectTimeoutMS, OnTimeElapsed));
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(chosenTransport));
+            }
+
+            Show();
+        }
+
+        public void Cancel()
+        {
+            End();
+        }
+
+        void End()
+        {
+            Hide();
+            StopAllCoroutines();
+        }
+
+        IEnumerator WaitUntilUNETDisconnected(Action endAction)
+        {
+            yield return new WaitUntil(() => !NetworkManager.Singleton.IsListening);
+
+            endAction();
+        }
+
+        IEnumerator DisplayUTPReconnectAttempts(int maxReconnectAttempts, int connectTimeoutMS, Action endAction)
```

Comment:
As discussed, let's just use time * nbAttempts for this countdown. We want to be SDK authoritative.